src/carousel.tsx

```diff
@@ -353,6 +355,11 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
     const showNextArrow = scrollRight !== 0;
     const showPrevArrow = scrollLeft !== 0;
 
+    const initialItemInfo = React.useRef({
+        initialItem: 0,
+        itemsPerPage: itemsPerPageFloor,
+    });
+
```

Comment:
required to assure the scroll is updated only when the prop is changed, but also when the items per page changes (to avoid leaving the carousel on a corrupt state).